Updated RDA addresses and dataset names

kdraeder opened the pull request:
## Description:
NCAR's Research Data Archive changed their dataset naming convention,
so all the references in DART became outdated.  
The "local" (/glade) versions were also moved to a new location (/glade/campaign) when derecho replaced cheyenne.

Did not update
   ./CHANGELOG.rst
   ./_build/html/CHANGELOG.html
   ./_build/html/_sources/CHANGELOG.rst.txt

These substitutions were made:
sed \\
    -e "s#glade\/collections#glade\/campaign\/collections#" \\
    -e "s#edu\/ds\/#edu\/datasets\/ds#g" \\
    -e "s#ds345.0#d345000#g" \\
    -e "s#ds199.1#d199001#g" \\
    -e "s#ds090.0#d090000#g" \\
    -e "s#ds337.0#d337000#g" \\
    -e "s#ds285.0#d285000#g" \

@hkershaw-brown I updated some .html files too, and would like confirmation that that was 
the right thing to do.

@braczka Please check some of the CLM files, but I think that you don't need check all of them.

I did not add DOIs for these datasets, which would be more robust, but more work.

### Fixes issue
#755

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Documentation changes needed?
<!-- Put an `x` in all the boxes that apply: -->
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.

### Tests
I checked the changes in 
./models/POP/shell_scripts/cesm2_1/user_datm.streams.txt.CPLHISTForcing.Solar_template

I did not test any of the testable files.  

## Checklist for merging

- [ ] Updated changelog entry
- [ ] Documentation updated
- [ ] Update conf.py

## Checklist for release
- [ ] Merge into main
- [ ] Create release from the main branch with appropriate tag
- [ ] Delete feature-branch

## Testing Datasets

- [ ] Dataset needed for testing available upon request
- [ ] Dataset download instructions included
- [ ] No dataset needed

hkershaw-brown commented:
> @hkershaw-brown I updated some .html files too, and would like confirmation that that was
the right thing to do.

Does not look like any html files in this pull request. Do you mean the .rst files?

kdraeder commented:
> > @hkershaw-brown I updated some .html files too, and would like confirmation that that was
> > the right thing to do.
> 
> Does not look like any html files in this pull request. Do you mean the .rst files?

Apparently the files in _build/html/... are not under git control.   I'm guessing those are exclude intentionally.

hkershaw-brown commented:
> Apparently the files in _build/html/... are not under git control. I'm guessing those are exclude intentionally.

Yes those are files from the local build of the documentation.

hkershaw-brown commented:
How about for the .rst  having one reference file for all the data sets so they only need to be changes in one place (when the rda links change, or change to dois).

.. CAM6rean :: https://rda.ucar.edu/datasets/d345000/

.. CAM4rean :: https://rda.ucar.edu/datasets/whatever

.. NCEPrda :: https://rda.ucar.edu/datasets/blahblah

kdraeder commented:
> How about for the .rst having one reference file for all the data sets so they only need to be changes in one place (when the rda links change, or change to dois).
> 
> .. CAM6rean :: https://rda.ucar.edu/datasets/d345000/
> 
> .. CAM4rean :: https://rda.ucar.edu/datasets/whatever
> 
> .. NCEPrda :: https://rda.ucar.edu/datasets/blahblah

That would tidy some of the files nicely.  But most of the files are stream files.
I'll work on it.

hkershaw-brown commented:
Add this to conf.py them you can have a guide/references.rst (or whatever you want to call it) included in every file. This means you can use `CAM6rean`_ anywhere in the docs.

```
 89 # include references
 90 with open(os.path.join(os.path.dirname(__file__), 'guide/references.rst')) as f:
 91     references_content = f.read()
 92 
 93 rst_prolog = references_content
````

kdraeder commented:
In my conf.py I needed to activate the os package to make the references lines work.

I also added `.. include:: ` lines into the rst files that need to use the references.rst file,
but Sphinx(?) is changing the path names in a way that I haven't parsed yet.
On my mac the full path name I give it gets converted to 
../Manhattan/Users/raeder/DAI/Manhattan/guide/references.rst
so it can't find definitions for the rda web site references.

hkershaw-brown commented:
@kdraeder do you want me to fix it or help you fix it?

hkershaw-brown commented:
I'll just go ahead and fix this Kevin

hkershaw-brown reviewed (approved):
I switched the https://rda.ucar.edu/datasets links for targets in guide/references.rst.

I'm happy for this to be merged and released. 

@braczka  do you have any objections?

kdraeder commented:
I'm still trying to figure out what you changed and how it compares with the work I did Friday, but didn't commit yet.   My git skills are not the quickest.
My list of references is longer, and I removed "anonymous" links, since they seem to be less robust.
If you want to be done with it, go ahead.  If you want to see my changes, then I need to figure out how to `include` the references.rst file into the files that need it.  So far I don't see an `include` in your cam-fv/readme.rst to use as a template.

> I switched the https://rda.ucar.edu/datasets links for targets in guide/references.rst.
> 
> I'm happy for this to be merged and released.
> 
> @braczka do you have any objections?

hkershaw-brown commented:
these line in the conf.py includes the guide/refrences.rst in very file 

# include references
with open(os.path.join(os.path.dirname(__file__), 'guide/references.rst')) as f:
    references_content = f.read()

rst_prolog = references_content

hkershaw-brown commented:
ok @kdraeder I guess you are doing the changes, I will redo the review.

hkershaw-brown commented on models/POP/readme.rst:
```diff
@@ -1,3 +1,7 @@
+.. Shouldn't be needed because of new os lines in conf.py:
+.. .. include:: /Users/raeder/DAI/Manhattan/guide/references.rst
+.. .. include:: ../../guide/references.rst
+
```
remove this, 1. not needed 2. hardcoded for your machine

hkershaw-brown commented on models/cam-fv/readme.rst:
```diff
@@ -1,3 +1,7 @@
+.. Shouldn't be needed because of new os lines in conf.py:
+.. .. include:: /Users/raeder/DAI/Manhattan/guide/references.rst
+.. .. include:: ../../guide/references.rst
+
```
remove this, same comment as previously

hkershaw-brown commented on models/cam-fv/readme.rst:
```diff
@@ -165,16 +171,16 @@ which contain the types of observations in the table below
 
 COSMIC
-   This site (http://www.cosmic.ucar.edu/) provides atmospheric refractivity 
+   `This site <http://www.cosmic.ucar.edu/>`_ provides atmospheric refractivity 
```
name the site rather 'COSMIC', or  www.cosmic.ucar.edu rather than 'this site'

kdraeder replied:
Good suggestion.  I was just making the minimal changes.

hkershaw-brown commented on models/cam-fv/readme.rst:
```diff
@@ -15,18 +19,19 @@ The DART system has supported data assimilation into CAM-FV continuously for man
 
 The variety of research can be sampled on the DART  
-`Publications <https://dart.ucar.edu/pages/Publications.html>`__ page.
+`Publications <https://dart.ucar.edu/pages/Publications.html>`_ page.
```
what's the goal of making this non-anonymous?

kdraeder replied:
In my limited reading I got the impression that anonymous is less robust, but now I see that it's an "embedded" URI and there are no other references to it (in this document), so anonymous is not any riskier than a named link or pointer.  I'll revert it.

hkershaw-brown commented on models/cam-fv/readme.rst:
```diff
@@ -6,7 +10,7 @@ Overview
 
 The larger context of the Community Earth System Model and DART interactions
 is described in the `CESM readme <../CESM/readme.html>`_
-This document focuses on the several `atmospheric models <http://www2.cesm.ucar.edu/models>`__
+This document focuses on the several `atmospheric models <http://www2.cesm.ucar.edu/models>`_
```
Are you updating all the links in this document?

If so, <../CESM/readme.html>`_ should be changed to not be html so it builds correctly with latex, etc. builds of the documentation.

kdraeder replied:
My goal was not to update all the links, but it seemed like I should fix link-related things that I noticed. My intent was the same as the Publications change (previous comment), but you've brought up a new issue - latex - so I'm wondering whether the html form is fine in an anonymous link (__) but not in a named link (_).

kdraeder replied:
I'd probably enjoy making it super tidy and maintainable, but I don't want to add that mission creep to this issue.  I've opted for the :doc: form (in 3 places).  Thanks for the tips!

hkershaw-brown commented on models/cam-fv/readme.rst:
```diff
@@ -15,18 +19,19 @@ The DART system has supported data assimilation into CAM-FV continuously for man
 
-.. |CAM6_Rean| replace:: 1 degree reanalysis wiki
-.. _CAM6_Rean: https://github.com/NCAR/DART/wiki/1-degree,-CAM6,-ensemble-reanalysis-for-CESM-experiments-(2011-thru-2019):-DATM,-hindcasts,-model-evaluation
+.. replaced by references.rst CAM6Rean_wiki
+   _CAM6_Rean: https://github.com/NCAR/DART/wiki/1-degree,-CAM6,-ensemble-reanalysis-for-CESM-experiments-(2011-thru-2019):-DATM,-hindcasts,-model-evaluation
```
this is just commented out now?  Just remove it.

hkershaw-brown commented on models/cam-fv/readme.rst:
```diff
@@ -118,8 +123,9 @@ Reanalyses
-`DS345.0 <https://rda.ucar.edu/datasets/ds345.0/#!description>`__ .
-(See the |CAM6_Rean|_ ).
+`d345000 <CAM6rean_rda_>`_ .
+(See the `CAM6rean_wiki`_ for details).
```
You've changed the name of the link here.  Main branch:
(See the 1 degree reanalysis wiki ).

This pull request:
(See the CAM6rean_wiki for details).

kdraeder replied:
That was a reference formatting error.  But the doc still refers to the wiki by different words.  Is that a problem if the rst name of the link is consistent?

hkershaw-brown replied:
no not a problem just checking that you were ok with having the link test be CAM6rean_wiki vs 1 degree reanalysis wiki. Either seems ok to me

hkershaw-brown commented on models/cam-fv/readme.rst:
```diff
@@ -154,7 +160,7 @@ It contains about 1.5Tb of data:
    stream text files are included in the RDA to illustrate how to use these data.
 
-..  * CAM4, 2 degree, 2000-2010, `Reanalysis <https://rda.ucar.edu/datasets/ds199.1>`__
+..  * CAM4, 2 degree, 2000-2010, `Reanalysis <https://rda.ucar.edu/datasets/d199001>`__
```
this is a comment.  Is it needed?

hkershaw-brown commented on models/cam-fv/readme.rst:
```diff
@@ -165,16 +171,16 @@ which contain the types of observations in the table below
+/glade/campaign/cisl/fs1_p_relocation/dares/Observations/NCEP+ACARS+GPS+AIRS/Thinned_x9x10.
 Versions of these files, which also have the results of the reanalysis in them,
-are available from the RDA ds345.0 linked above.
+are available from the RDA d345000 linked above.
```
why not link here?

kdraeder replied:
When I first wrote it they were closer, and the extra typing didn't seem worthwhile.  I've made it a link now.

hkershaw-brown reviewed (approved):
Nice work Kevin!

kdraeder commented:
> Nice work Kevin!

Thanks!  and thanks for your suggestions.
